chia/consensus/blockchain.py

```diff
@@ -286,17 +288,34 @@ async def receive_block(
         if fork_height is not None:
             # new coin records added
             assert coin_record_change is not None
-            return ReceiveBlockResult.NEW_PEAK, None, fork_height, coin_record_change
+            return ReceiveBlockResult.NEW_PEAK, None, fork_height, (coin_record_change, hint_changes)
         else:
-            return ReceiveBlockResult.ADDED_AS_ORPHAN, None, None, []
+            return ReceiveBlockResult.ADDED_AS_ORPHAN, None, None, ([], {})
+
+    def get_hint_list(self, npc_result: NPCResult) -> List[Tuple[str, str]]:
+        h_list = []
+        for npc in npc_result.npc_list:
+            for opcode, conditions in npc.conditions:
+                if opcode == ConditionOpcode.CREATE_COIN:
+                    for condition in conditions:
+                        if len(condition.vars) > 2:
```

Comment:
the next version of `clvm_rs` will actually always include the 3rd field here, but it will be an empty bytes buffer for coins that didn't specify it. So you'll need to check `condition.vars[2] != b""` as well